### PR TITLE
[ENG-2496] Replace validated models in settings.account route

### PR DESCRIPTION
**Pull request by futa-ikeda**
- Ticket: [ENG-2496]
- Feature flag: n/a

## Purpose
- Replace validated-model-form with form-controls

## Summary of Changes
- Replaced validated model form in Settings::Account::-Components::ConnectedEmails
- Replaced validated model form in Settings::Account::-Components::DefaultRegion
- Replaced validated model form in Settings::Account::-Components::Security

## Side Effects
NA

## QA Notes
- The connected emails panel (adding a new email for the user), the default storage location panel, and security settings panel on osf.io/settings/account _should_ function the same, but a quick spot check is always appreciated! 

[ENG-1528]: https://openscience.atlassian.net/browse/ENG-1528

[ENG-2496]: https://openscience.atlassian.net/browse/ENG-2496

### CODE REVIEW

File: app/settings/account/-components/connected-emails/component.ts [hunk 1]
```diff
@@ -9,23 +12,35 @@ import Toast from 'ember-toastr/services/toast';
...
+interface EmailValidation {
+    emailAddress: string;
+}
+
+const emailValidations: ValidationObject<EmailValidation> = {
+    emailAddress: [
+        validateFormat({
+            allowBlank: false,
+            type: 'email',
+            translationArgs: { description: 'Email address' },
```
Comment (brianjgeiger) [inline app/settings/account/-components/connected-emails/component.ts:28]: Drive-by: should this be a `t()` string rather than a bare English string?

File: app/settings/account/-components/default-region/component.ts [hunk 2]
```diff
@@ -12,15 +14,32 @@ import Toast from 'ember-toastr/services/toast';
...
+interface RegionValidation {
+    defaultRegion: RegionModel;
+}
+
+const regionValidation: ValidationObject<RegionValidation> = {
+    defaultRegion: [
+        validatePresence({
+            presence: true,
+            ignoreBlank: true,
+            allowBlank: false,
```
Comment (fabmiz) [inline app/settings/account/-components/default-region/component.ts:29]: haven't gone through the validatePresence API yet but don't these two contradict?

File: app/settings/account/-components/default-region/component.ts [hunk 3]
```diff
@@ -37,40 +56,39 @@ export default class DefaultRegionPane extends Component {
...
-
+        this.set('user', user);
+        this.changeset = buildChangeset(user, regionValidation, { skipValidate: true });
         yield user.belongsTo('defaultRegion').reload();
     });
 
+    @task({ withTestWaiter: true })
+    updateRegion = task(function *(this: DefaultRegionPane) {
+        this.changeset.validate();
+        if (this.changeset.get('isValid') && this.user) {
```
Comment (fabmiz) [inline app/settings/account/-components/default-region/component.ts:67]: Can we avoid using `.get` whenever possible?
```suggestion
        if (this.changeset.isValid && this.user) {
```

**Review by fabmiz — COMMENTED** [review]
First pass: overall looks pretty good.

File: app/settings/account/-components/default-region/component.ts (hunk 2 above)
Comment (futa-ikeda) [reply app/settings/account/-components/default-region/component.ts:29]: `ignoreBlank: true` will ignore empty and whitespace. I think allowBlank and ignoreBlank isn't useful for the default region though, since it's a power-selector, and shouldn't have any empty/blank inputs

File: app/settings/account/-components/connected-emails/component.ts [hunk 4]
```diff
@@ -121,34 +136,35 @@ export default class ConnectedEmails extends Component {
...
+            if (this.changeset.get('isValid') && this.changeset.get('emailAddress')) {
+                this.set('lastUserEmail', this.changeset.get('emailAddress'));
+                newEmail = this.store.createRecord('user-email', {
+                    emailAddress: this.changeset.get('emailAddress'),
+                    user: this.currentUser.user,
+                });
+                yield newEmail.save();
+                this.set('showAddModal', true);
+                this.reloadUnconfirmedList();
+                this.toast.success(this.intl.t('settings.account.connected_emails.save_success'));
```
Comment (adlius) [inline app/settings/account/-components/connected-emails/component.ts:153]: Should probably clear the changeset value after successfully saving the email, otherwise the input box would still have the added email in it.

**Review by adlius — CHANGES_REQUESTED** [review]
One minor thing. Otherwise I have tested all functionalities locally and it works. The code looks clean too.

**Review by adlius — APPROVED** [review]

File: app/settings/account/-components/security/component.ts [hunk 5]
```diff
@@ -75,6 +104,26 @@ export default class SecurityPane extends Component {
         }
     });
 
+    @task({ withTestWaiter: true })
+    verifySecret = task(function *(this: SecurityPane) {
+        this.changeset.validate();
+        try {
+            if (this.changeset.isValid) {
+                yield this.changeset.save({});
+                this.showError = false;
```
Comment (fabmiz) [inline app/settings/account/-components/security/component.ts:113]: I think this should be done first thing in this task? At least it looks like that's how it used to work.

**Review by fabmiz — CHANGES_REQUESTED** [review]
Need to get rid of [model validations for `user-email`](https://github.com/CenterForOpenScience/ember-osf-web/blob/0b15078667814998cb9a5954bcc077f8612daed2/app/models/user-email.ts#L15-L38)

File: app/settings/account/-components/security/component.ts (hunk 5 above)
Comment (futa-ikeda) [reply app/settings/account/-components/security/component.ts:113]: I think the old `verifySecret` action was called when the save was successful, so I _think_ it still belongs in this block right here, after the changeset is saved. My understanding of ValidatedModelForm isn't the best though

**Review by fabmiz — APPROVED** [review]
🥇
